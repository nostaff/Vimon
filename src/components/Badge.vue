<template>
  <span v-text="text" :class="['badge', {'badge-dot': typeof text === 'undefined', 'badge-single': typeof text !== 'undefined' && text.toString().length === 1}]"></span>
</template>

<script>
export default {
  props: {
    text: [String, Number]
  }
}
</script>
