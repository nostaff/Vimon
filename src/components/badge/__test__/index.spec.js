/* eslint-disable no-undef,no-unused-expressions */

import { mount } from 'vue-test-utils'
import deepAssign from 'deep-assign'
import cloneDeep from 'lodash.clonedeep'
import Badge from '../index'
import { createRenderer } from 'vue-server-renderer'

let options = {
  propsData: {
    color: 'primary',
    theme: 'ios'
  },
  slots: {
    default: '<span>Test</span>'
  }
}

describe('Badge', function () {
  // 清除DOM痕迹
  it('@base: renders the correct markup', function () {
    let opts = deepAssign(cloneDeep(options), {
      propsData: {
        theme: 'md'
      }
    })
    let wrapper = mount(Badge, opts)
    const result = '<span class="ion-badge badge badge-md badge-md-primary"><span>Test</span></span>'
    expect(wrapper.html()).toEqual(result)
  })

  it('@base: renders the correct text', function () {
    let opts = deepAssign(cloneDeep(options), {
      slots: {
        default: '<span>Test</span>'
      }
    })
    let wrapper = mount(Badge, opts)
    expect(wrapper.text()).toEqual('Test')
  })

  it('@base: component must have a name', function () {
    let opts = deepAssign(cloneDeep(options), {
      slots: {
        default: '<span>Name</span>'
      }
    })
    let wrapper = mount(Badge, opts)
    expect(wrapper.name()).toEqual('ion-badge')
  })

  it('@base: have the right className', function () {
    let opts = deepAssign(cloneDeep(options), {
      slots: {
        default: '<span>HasClass</span>'
      }
    })
    let wrapper = mount(Badge, opts)
    expect(wrapper.hasClass('ion-badge')).toBeTruthy()
  })

  it('@base: matches snapshot', () => {
    const renderer = createRenderer()
    const wrapper = mount(Badge, options)
    renderer.renderToString(wrapper.vm, (err, str) => {
      if (err) throw new Error(err)
      expect(str).toMatchSnapshot()
    })
  })

  it('@props: color', function () {
    let opts = deepAssign(cloneDeep(options), {
      propsData: {
        color: 'danger',
        theme: 'md'
      },
      slots: {
        default: '<span>color</span>'
      }
    })
    let wrapper = mount(Badge, opts)
    expect(wrapper.hasClass('badge-md-danger')).toBeTruthy()
  })

  it('@props: theme', function () {
    let opts = deepAssign(cloneDeep(options), {
      propsData: {
        theme: 'md'
      },
      slots: {
        default: '<span>theme</span>'
      }
    })
    let wrapper = mount(Badge, opts)
    expect(wrapper.hasClass('badge-md')).toBeTruthy()
  })

  it('@props: theme(default)', function () {
    let wrapper = mount(Badge)
    expect(wrapper.hasClass('badge-ios')).toBeTruthy()
  })
})
