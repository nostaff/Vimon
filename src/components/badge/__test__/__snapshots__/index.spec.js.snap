// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Badge @base: matches snapshot 1`] = `<span class="ion-badge badge badge-ios badge-ios-primary"><span>Test</span></span>`;
